### DISCUSSION TITLE
How to select library when external dependency provides multiple

### DISCUSSION THREAD

**Discussion by vid512** [mesonbuild/meson#11298] [Q&A] [answered]
In my project I use catch2, version 3.

```
catch2_dep = dependency('catch2', version: '>=3')
...
executable(..., dependencies: [catch2_dep, ...])
```

This library is built in two versions: with main, and without main. 
```
$ ls build/subprojects/Catch2-3.2.0/src/catch2/*.a
build/subprojects/Catch2-3.2.0/src/catch2/libCatch2.a
build/subprojects/Catch2-3.2.0/src/catch2/libCatch2Main.a
```

Meson builds both, but links my executable with libCatch2.a by default. I want to link with libCatch2Main.a. How can I instruct Meson to pick the other one?

**Comment by vid512** [comment]
More info on how this is done in CMake, in case it would be somehow helpful:

https://github.com/catchorg/Catch2/blob/devel/docs/cmake-integration.md

> Catch2's CMake build exports two targets, Catch2::Catch2, and Catch2::Catch2WithMain. If you do not need custom main function, you should be using the latter (and only the latter). Linking against it will add the proper include paths and link your target together with 2 static libraries that implement Catch2 and its main respectively. If you need custom main, you should link only against Catch2::Catch2.

**Comment by eli-schwartz** [comment, marked as answer]
Catch2 provides a catch2-with-main.pc, so try `dependency('catch2-with-main')`.

**Comment by vid512** [comment]
It works. Thank you.
